bench/src/main/scala/bench/GenBench.scala

```diff
@@ -0,0 +1,104 @@
+package bench
+
+import java.util.concurrent.TimeUnit
+import org.openjdk.jmh.annotations._
+
+import org.scalacheck.Gen
+import org.scalacheck.rng.Seed
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+class GenBench {
+
+  // if we only benchmark with one seed, we might get lucky (or
+  // unlucky) for a particular generator. by benchmarking with 10
+  // different seeds (constant between runs) we reduce the chance of
+  // extreme luckiness or unluckness. that said, a generator that hits
+  // a 100x slow down 1% of the time will likely cause problems for
+  // this benchmarking strategy.
+  val preseeds: List[Long] =
+    List(
+      -4522249856328521867L, 624585044422885589L, 3202652801110979260L,
+      -8551572270962952301L, 2834982173645900841L, -7161929398350763557L,
+      -249952150141786145L, -7514897450101711682L, 9111269543303992388L,
+      -8189142734131299482L)
+
+  val seeds: List[Seed] =
+    preseeds.map(n => Seed(n))
+
+  val params: Gen.Parameters =
+    Gen.Parameters.default
+
+  val genInt: Gen[Int] =
+    Gen.choose(Int.MinValue, Int.MaxValue)
+
+  @Benchmark
+  def const_(): List[Long] =
+    seeds.map(s => Gen.const(999L).pureApply(params, s))
```

Comment:
Should the benchmarks have a parameter for a [blackhole](http://tutorials.jenkov.com/java-performance/jmh.html#passing-value-to-a-black-hole) and consume any return values?  

I'm not sure whether it's a concern for Scala code or not, but it seems cargo cult worth adopting.

Reply:
I'm pretty sure the Scala plugin does this automatically. When I looked at `/Users/erik/w/scalacheck/bench/target/scala-2.13/src_managed/jmh/bench/generated/GenBench_listOfInt_jmhTest.java` I see `Blackhole` values being instantiated and consuming the benchmark results.

Reply:
Excellent! Good to know.